sdk/include/opentelemetry/sdk/metrics/aggregation/aggregation.h

```diff
@@ -11,24 +11,42 @@ namespace sdk
 {
 namespace metrics
 {
-class InstrumentMonotonicityAwareAggregation
-{
-public:
-  InstrumentMonotonicityAwareAggregation(bool is_monotonic) : is_monotonic_(is_monotonic) {}
-  bool IsMonotonic() { return is_monotonic_; }
-
-private:
-  bool is_monotonic_;
-};
-
 class Aggregation
 {
 public:
   virtual void Aggregate(long value, const PointAttributes &attributes = {}) noexcept = 0;
 
   virtual void Aggregate(double value, const PointAttributes &attributes = {}) noexcept = 0;
 
-  virtual PointType Collect() noexcept = 0;
+  /**
+   * Returns the result of the merge of the two aggregations.
+   *
+   * This should always assume that the aggregations do not overlap and merge together for a new
+   * cumulative report.
+   *
+   * @param delta the newly captured (delta) aggregation
+   * @return the result of the merge of the given aggregation.
+   */
+
+  virtual std::unique_ptr<Aggregation> Merge(Aggregation &delta) noexcept = 0;
```

Comment:
Could the new virtual function be marked as `const`?

Reply:
Yes, have marked Merge, Diff and ToPoint as `const`